### PR TITLE
Remove concrete table inheritance

### LINKED ISSUE #7371: Documentation of Concrete Inheritance
The documentation of Concrete Inheritance when you use one table per class seems to be missing.

I only figured out you could use InheritanceType TABLE_PER_CLASS by looking at the source code.

### PR DESCRIPTION
**Pull request by mpdude**
This seems to be a leftover from early days, but never actually implemented or documented.

#10423 adds the deprecation notices in 2.14.

Closes #7371, references #10217, closes #10220.

### CODE REVIEW

**Review by greg0ire — APPROVED** [review]
Not implemented in a decade. Let's indeed clean this up 👍

**Comment by mpdude** [comment]
Since it technically a BC break, do we require deprecation warnings or similar in 2.x?

**Comment by greg0ire** [comment]
It can't hurt to do so. If you do, you'll have to add something in `UPGRADE.md` about this, and also, you will have to edit `UPGRADE.md` in this PR with `s/deprecated/removed/`

**Comment by mpdude** [comment]
I don't know how access to a `const` could be deprecated, and where can I find the pattern that this project uses to announce deprecations (e. g. a `@trigger_error(...)` or so?)

**Comment by greg0ire** [comment]
For once, we have a clearly documented policy on this: https://www.doctrine-project.org/policies/deprecation.html :slightly_smiling_face: 

TL;DR: use `Deprecation::trigger()`, and when you can't, use `@deprecated`

**Review by SenseException — APPROVED** [review]

**Review by derrabus — APPROVED** [review]
